src/python/pants/bin/remote_pants_runner.py

```diff
@@ -114,15 +118,19 @@ def run(self, start_time: float) -> ExitCode:
         pantsd_handle = self._client.maybe_launch()
         logger.debug(f"Connecting to pantsd on port {pantsd_handle.port}")
 
-        return self._connect_and_execute(pantsd_handle, start_time)
+        executor = GlobalOptions.create_py_executor(self._bootstrap_options.for_global_scope())
+        try:
+            return self._connect_and_execute(pantsd_handle, executor, start_time)
+        finally:
+            executor.shutdown(3)
 
     def _connect_and_execute(
-        self, pantsd_handle: PantsDaemonClient.Handle, start_time: float
+        self,
+        pantsd_handle: PantsDaemonClient.Handle,
+        executor: PyExecutor,
+        start_time: float,
     ) -> ExitCode:
         global_options = self._bootstrap_options.for_global_scope()
-        # We do not explicitly shut this PyExecutor down, because the client should not run any long lived
-        # tasks which we would want to wait for (in particular: it runs no Python code). See #16105.
-        executor = GlobalOptions.create_py_executor(global_options)
```

Comment:
This wasn't actually accurate because of this code: https://github.com/pantsbuild/pants/blob/2983f166e18b1c1a480b85bf722d16d514b41b17/src/rust/engine/src/externs/scheduler.rs#L30-L41